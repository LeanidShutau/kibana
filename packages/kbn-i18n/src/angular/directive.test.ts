/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import angular from 'angular';
import 'angular-mocks';
import 'angular-sanitize';

import { i18nDirective } from './directive';
import { I18nProvider } from './provider';

angular
  .module('app', ['ngSanitize'])
  .provider('i18n', I18nProvider)
  .directive('i18nId', i18nDirective);

describe('i18nDirective', () => {
  let compile: angular.ICompileService;
  let scope: angular.IRootScopeService;

  beforeEach(angular.mock.module('app'));
  beforeEach(
    angular.mock.inject(
      ($compile: angular.ICompileService, $rootScope: angular.IRootScopeService) => {
        compile = $compile;
        scope = $rootScope.$new();
      }
    )
  );

  test('inserts correct translation html content', () => {
    const element = angular.element(
      `<div
        i18n-id="id"
        i18n-default-message="Default message"
      />`
    );

    compile(element)(scope);
    scope.$digest();

    expect(element[0]).toMatchSnapshot();
  });

  test('sanitizes defaultMessage', () => {
    const element = angular.element(
      `<div
        i18n-id="id"
        i18n-default-message="Dangerous default message, {value}"
        i18n-values="{ value: '<div i18n-id=&quot;id2&quot; i18n-default-message=&quot;<script></script>inner message&quot; />' }"
      />`
    );

    compile(element)(scope);
    scope.$digest();

    expect(element[0]).toMatchSnapshot();
  });

  test('inserts correct translation html content with values', () => {
    const element = angular.element(
      `<div
        i18n-id="id"
        i18n-default-message="Default message, {value}"
        i18n-values="{ value: '<div i18n-id=&quot;id2&quot; i18n-default-message=&quot;<span>inner message</span>&quot; />' }"
      />`
    );

    compile(element)(scope);
    scope.$digest();

    expect(element[0]).toMatchSnapshot();
  });
});
