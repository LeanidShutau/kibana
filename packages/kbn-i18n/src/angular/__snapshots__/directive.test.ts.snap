// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i18nDirective inserts correct translation html content 1`] = `
<div
  class="ng-scope ng-isolate-scope"
  i18n-default-message="Default message"
  i18n-id="id"
>
  Default message
</div>
`;

exports[`i18nDirective inserts correct translation html content with values 1`] = `"default-message word"`;

exports[`i18nDirective inserts correct translation html content with values 2`] = `"default-message anotherWord"`;

exports[`i18nDirective sanitizes defaultMessage 1`] = `
<div
  class="ng-scope ng-isolate-scope"
  i18n-default-message="Dangerous default message, {value}"
  i18n-id="id"
  i18n-values="{ value: '<div i18n-id=\\"id2\\" i18n-default-message=\\"<script></script>inner message\\" />' }"
>
  Dangerous default message, 
  <div
    class="ng-scope ng-isolate-scope"
    i18n-default-message="<script></script>inner message"
    i18n-id="id2"
  >
    inner message
  </div>
</div>
`;
